MediaGalleryIntegration/Test/Unit/Plugin/UpdateOpenDialogUrlTest.php

```diff
@@ -66,9 +66,19 @@ protected function setUp(): void
      */
     public function testAfterPrepareWhenConfigDisabled(): void
     {
-        $this->configMock->expects($this->once())->method('isEnabled')->willReturn(false);
-        $this->imageComponentMock->expects($this->never())->method('setData');
-        $this->urlMock->expects($this->never())->method('getUrl');
+        $this->configMock->expects($this->once())
+            ->method('isEnabled')
+            ->willReturn(false);
+
+        $this->imageComponentMock->expects($this->never())
+            ->method('getData');
+
+        $this->urlMock->expects($this->never())
+            ->method('getUrl');
+
+        $this->imageComponentMock->expects($this->never())
```

Comment:
Such strict assertions make test more fragile, why need to add them?

Reply:
As I understood this was added to check that if the https://github.com/magento/adobe-stock-integration/blob/29acaec609d4d760cb00176d9bdd837824d2093d/MediaGalleryIntegration/Plugin/UpdateWysiwygOpenDialogUrl.php#L59 returns false no other execution will be not fired from the code perspective.

Reply:
In such case I would recommend asserting `$config` remain unchanged instead. Try to rely on behavior rather than on execution flow in tests.

Reply:
@orlangur thank you for your review and suggestions. This test implemented like this because of the method returns `void` Please, advise how can I check that Image object which is a plugin param here doesn't have new data such as `['config']['mediaGallery']['openDialogUrl']`?

`$this->imageComponentMock->expects($this->never())` is a kind of behaviour: if config false we do not execute it and this means that we do not change original object.

Reply:
I think it's ok to assert that there were no calls to `imageComponentMock` for a unit test, however, I don't see the point to mock a return value in this case